.github/workflows/sam-pipeline.yml

```diff
@@ -0,0 +1,63 @@
+name: SAM Validate, Build, Test, Deploy
+on:
+  push:
+    branches:
+      - main
+      - feature/use-aws
```

Comment:
@hieu-aiven Can we improve the workflow to trigger on file changes in the `aws` folder only? Reference for [GitHub Actions triggers](https://www.techielass.com/trigger-a-github-action-workflow-when-a-folder-changes/)